### PR TITLE
[15 min fix] Fallback to the image URL if Cloudinary/Imgproxy are not configured

### PR DESCRIPTION
**Pull request by rhymes**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Currently we tell people to copy `.env_sample` to `.env` but if they then try to link an external image (without uploading it) in a post, the `Images::Optimizer` will try to generate a Cloudinary URL and leave the user with a broken UX.

I think `Images::Optimizer` should just return the image URL verbatim if neither Cloudinary nor Imgproxy are configured and that we should have that as a default for new installs.

## QA Instructions, Screenshots, Recordings

1. set `CLOUDINARY_API_KEY` and `CLOUDINARY_CLOUD_NAME` back to `Optional` (or anything really), start the app, get a [image URL](https://media1.tenor.com/images/38764a3868c2dd4f8c1253c4e9ec896e/tenor.gif) and create a post with:

```markdown
![image](https://media1.tenor.com/images/38764a3868c2dd4f8c1253c4e9ec896e/tenor.gif)
```

2. Cloudinary will create a broken URL because
3. empty `CLOUDINARY_API_KEY` and `CLOUDINARY_CLOUD_NAME` again and restart the app, re-save the previous article or create a new one, now you'll see the image directly from where you've linked it.

## Added tests?

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

### CODE REVIEW

**Comment by citizen428** [comment]
Seems like there are quite a few specs failing. I also noticed that the Cloudinary initializer sets a name for the test env, maybe you want to change that too:

```ruby
  config.cloud_name = Rails.env.test? ? "TEST-CLOUD" : ApplicationConfig["CLOUDINARY_CLOUD_NAME"]
```

**Comment by rhymes** [comment]
Yeah, a lot of specs assume Cloudinary to be always there. I'll address them. I haven't changed that because I assumed `TEST-CLOUD` was something existing :D I'll take a look at that too, thank you!

File: spec/models/message_spec.rb [hunk 1]
```diff
@@ -43,6 +43,18 @@
 
     describe "#message_html" do
       it "creates rich link with proper link for article" do
+        link = "http://#{ApplicationConfig['APP_DOMAIN']}#{article.path}"
```
Comment (citizen428) [inline spec/models/message_spec.rb:46]: Would it make sense to use our URL helper here?

```suggestion
        link = URL.article(article)
```

**Review by citizen428 — APPROVED** [review]

**Review by maestromac — APPROVED** [review]
LGTM! TY for the extensive test!

**Pull request merged**